src/firmware_logger_device.cpp

```diff
@@ -44,91 +66,151 @@ namespace librealsense
 
     void firmware_logger_device::get_fw_logs_from_hw_monitor()
     {
-        auto res = _hw_monitor->send(_fw_logs_command);
-        if (res.empty())
+        command update_command = get_update_command();
+        if( update_command.cmd == 0 )
+            return;
+
+        auto res = _hw_monitor->send( update_command );
+        if( res.empty() )
         {
             return;
         }
 
-        auto beginOfLogIterator = res.begin();
-        // convert bytes to fw_logs_binary_data
-        for (int i = 0; i < res.size() / fw_logs::BINARY_DATA_SIZE; ++i)
+        // Convert bytes to fw_logs_binary_data
+        auto beginOfLogIterator = res.data();
+        size_t size_left = res.size();
+        while( size_left > 0 )
         {
-            auto endOfLogIterator = beginOfLogIterator + fw_logs::BINARY_DATA_SIZE;
-            std::vector<uint8_t> resultsForOneLog;
-            resultsForOneLog.insert(resultsForOneLog.begin(), beginOfLogIterator, endOfLogIterator);
-            fw_logs::fw_logs_binary_data binary_data{ resultsForOneLog };
-            _fw_logs.push(binary_data);
+            size_t log_size = get_log_size( beginOfLogIterator );
+            if( log_size > size_left )
+            {
+                LOG_WARNING( "Received an incomplete FW log" ); // TODO - remove after debugging, or decrease to debug level.
+                break;
+            }
+            auto endOfLogIterator = beginOfLogIterator + log_size;
+            fw_logs::fw_logs_binary_data binary_data;
+            binary_data.logs_buffer.assign( beginOfLogIterator, endOfLogIterator );
+            _fw_logs.push( std::move( binary_data ) );
             beginOfLogIterator = endOfLogIterator;
+            size_left -= log_size;
         }
     }
 
-    void firmware_logger_device::get_flash_logs_from_hw_monitor()
+    size_t firmware_logger_device::get_log_size( uint8_t * buff ) const
```

Comment:
Can the input be const? It does not seem that you are modifying it in this method

Reply:
Done.